### PR TITLE
feat: added ratelimits package

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: packages/ratelimits/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Change Log
```
Comment (Favna) [inline packages/ratelimits/CHANGELOG.md:1]: Nuke this file, Lerna autogenerates it when needed.

File: packages/ratelimits/README.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
+<div align="center">
+
+![Sapphire Logo](https://cdn.skyra.pw/gh-assets/sapphire.png)
+
+# @sapphire/ratelimits
+
+\**Deconstruct and generate snowflake IDs using BigInts.*i
```
Comment (Favna) [inline packages/ratelimits/README.md:7]: `\**` ??
`*i` ??

Also the description is wrong anyway

File: packages/ratelimits/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+	displayName: 'unit test',
+	preset: 'ts-jest',
+	testEnvironment: 'node',
+	testRunner: 'jest-circus/runner',
+	testMatch: ['<rootDir>/tests/**/*.test.ts'],
+	setupFilesAfterEnv: ['<rootDir>/tests/jest.setup.ts'],
```
Comment (Favna) [inline packages/ratelimits/jest.config.js:7]:
```suggestion
```

File: packages/ratelimits/package.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{
+	"name": "@sapphire/ratelimits",
+	"version": "1.0.1",
+	"description": "Deconstructs and generates snoflake IDs using BigInts",
```
Comment (Favna) [inline packages/ratelimits/package.json:4]: Update this

File: packages/ratelimits/package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{
+	"name": "@sapphire/ratelimits",
+	"version": "1.0.1",
```
Comment (Favna) [inline packages/ratelimits/package.json:3]:
```suggestion
	"version": "0.0.1",
```

version will be set to 1.0.0 by lerna

File: packages/ratelimits/rollup.config.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		cleaner({
+			targets: ['./dist/']
+		}),
+		typescript({ tsconfig: resolveDir(__dirname, 'src', 'tsconfig.json') }),
+		terser({
+			ecma: 2019,
+			// This will ensure that whenever Rollup is in watch (dev) mode, console logs will not be removed
+			// eslint-disable-next-line @typescript-eslint/naming-convention
+			compress: { drop_console: !Reflect.has(process.env, 'ROLLUP_WATCH') },
+			output: { comments: false }
```
Comment (Favna) [inline packages/ratelimits/rollup.config.ts:38]:
```suggestion
			format: { comments: false }
```

output is from old version of Terser

**Review by Favna — CHANGES_REQUESTED** [review]
- Update the list of files to modify in all-contributors file
- Write actual README

File: packages/ratelimits/src/lib/Bucket.ts [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+/**
+ * The bucket limits.
+ */
+export interface BucketLimit {
+	/**
+	 * The time between tickets.
+	 *
+	 * If timespan is zero, there will be no spacing between usages, this means
+	 * that a 5/5s bucket ({@link BucketLimit#limit Limit} of 5 in 5 seconds
```
Comment (vladfrangu) [inline packages/ratelimits/src/lib/Bucket.ts:23]:
```suggestion
	 * that a 5/5s bucket ({@link BucketLimit#limit limit} of 5 in 5 seconds
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by Favna — APPROVED** [review]

**Pull request merged**